src/rules/no-identical-conditions.ts

```diff
@@ -77,6 +81,34 @@ const rule: TSESLint.RuleModule<string, string[]> = {
           }
         }
       },
+      SwitchStatement(node: TSESTree.Node) {
+        const switchStmt = node as TSESTree.SwitchStatement;
+        const previousTests: TSESTree.Expression[] = [];
+        for (const switchCase of switchStmt.cases) {
+          if (switchCase.test) {
+            const { test } = switchCase;
+            const duplicateTest = previousTests.find(previousTest =>
+              areEquivalent(test, previousTest, sourceCode),
+            );
+            if (duplicateTest) {
+              report(
+                context,
+                {
+                  messageId: 'duplicatedCase',
+                  data: {
+                    line: duplicateTest.loc.start.line,
+                  },
+                  node: test,
+                },
+                [issueLocation(duplicateTest.loc, duplicateTest.loc, 'Original')],
+                duplicatedCaseMessage,
+              );
+            } else {
+              previousTests.push(test);
```

Comment:
What about adding a test case that shows that when you have more than 1 duplicate, they both point to the same first "duplicate"?